aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -503,47 +515,28 @@ public void query(
                     return;
                 }
 
-                LOG.debug("Deleting item in table: " + sqliteTable.getName() +
-                    " identified by ID: " + item.getId());
-
-                // delete always checks for ID first
-                final QueryPredicateOperation<?> idCheck =
-                    QueryField.field(primaryKeyName).eq(item.getId());
-                final QueryPredicate condition = !QueryPredicates.all().equals(predicate)
-                    ? idCheck.and(predicate)
-                    : idCheck;
-                final SqlCommand sqlCommand = sqlCommandFactory.deleteFor(modelSchema, condition);
-                if (sqlCommand.sqlStatement() == null || !sqlCommand.hasCompiledSqlStatement()) {
-                    onError.accept(new DataStoreException(
-                        "No delete statement found for the Model: " + modelSchema.getName(),
-                        AmplifyException.TODO_RECOVERY_SUGGESTION
-                    ));
-                    return;
-                }
-
-                synchronized (sqlCommand.getCompiledSqlStatement()) {
-                    final SQLiteStatement compiledSqlStatement = sqlCommand.getCompiledSqlStatement();
-                    compiledSqlStatement.clearBindings();
-                    bindStatementToValues(sqlCommand, null);
-                    // executeUpdateDelete returns the number of rows affected.
-                    final int rowsDeleted = compiledSqlStatement.executeUpdateDelete();
-                    compiledSqlStatement.clearBindings();
-                    if (rowsDeleted == 0) {
-                        throw new DataStoreException(
-                            "Failed to meet condition. Model was not deleted.",
-                            "Please verify the current state of saved item."
-                        );
+                // Use ModelTreeHelper to identify the models affected by cascading delete.
+                Map<ModelSchema, Set<String>> modelFamilyTree =
+                        sqLiteModelTree.descendantsOf(Collections.singleton(item));
+
+                for (ModelSchema schema : modelFamilyTree.keySet()) {
+                    for (String id : modelFamilyTree.get(schema)) {
+                        // Publish DELETE mutation for each affected item.
+                        String dummyJson = String.format("{\"id\":%s}", id);
+                        Model dummyItem = gson.fromJson(dummyJson, schema.getModelClass());
+                        itemChangeSubject.onNext(StorageItemChange.builder()
+                                .changeId(id)
```

Comment:
Unrelated to this PR, but this seems wrong.  Shouldn't we not set `changeId` to anything here, so that `StorageItemChange::build` just randomly generates one?  From the `changeId` Javadoc, this is supposed to be a UUID representing this change, not the id of the model that it relates to.

Maybe this is something that we could follow up on in a separate PR.

Reply:
hmm, i also noticed that but it seems that other operations seem to treat change ID as model ID as well. I'll leave it as is for now and we can revisit it later as you said.